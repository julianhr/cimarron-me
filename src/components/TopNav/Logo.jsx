import React from 'react'
import PropTypes from 'prop-types'
import styled from '@emotion/styled'

import config from '~/config'
import AnchorTag from '~/library/AnchorTag'


const Root = styled.div`
  display: flex;
  flex-direction: column;
`

const H3 = styled.h3`
  color: ${props => props.theme.colors.text};
  font-size: 20px;
  font-weight: 400;
  margin: 0;
`

const H6 = styled.h6`
  padding-top: 2px;
  font-size: 11px;
  font-weight: 300;
  margin: 0;
  color: #999;
`

function Logo({ isLinkRouted }) {
  return (
    <Root>
      <AnchorTag
        url='/'
        label='Home'
        rootStyle={{ textDecoration: 'none' }}
        isLinkRouted={isLinkRouted}
      >
        <H3>{config.author.name}</H3>
        <H6>{config.author.title}</H6>
      </AnchorTag>
    </Root>
  )
}

Logo.propTypes = {
  isLinkRouted: PropTypes.bool,
}

export default Logo
