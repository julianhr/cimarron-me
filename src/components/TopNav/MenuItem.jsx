import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import styled from '@emotion/styled'


const Root = styled.div`
  height: 100%;
  display: flex;
  align-items: center;
`

const StyledLink = styled(Link)`
    color: ${props => props.theme.colors.text};
    text-decoration: none;
    margin: 0 10px;
    padding: 0 6px 4px;
    border-bottom: 2px solid transparent;
    transition: border-bottom-color 75ms linear;

    :hover {
      border-bottom-color: #e10d75;
      transition-duration: 140ms;
    }
`

function MenuItem(props) {
  return (
    <Root>
      <StyledLink
        to={props.url}
      >
        {props.label}
      </StyledLink>
    </Root>
  )
}

MenuItem.propTypes = {
  url: PropTypes.string,
  label: PropTypes.string,
}

export default MenuItem
