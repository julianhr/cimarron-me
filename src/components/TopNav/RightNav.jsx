import React from 'react'
import PropTypes from 'prop-types'
import styled from '@emotion/styled'

import MenuItem from './MenuItem'


const Root = styled.div`
  display: flex;
  align-items: flex-end;
`

function RightNav(props) {

  const renderMenuItems = menuItems => (
    menuItems.map((item, i) => (
      <MenuItem
        key={i}
        url={item.url}
        label={item.label}
      />
    ))
  )

  return (
    <Root>
      {renderMenuItems(props.menuItems)}
    </Root>
  )
}

RightNav.propTypes = {
  menuItems: PropTypes.array,
}

export default RightNav
