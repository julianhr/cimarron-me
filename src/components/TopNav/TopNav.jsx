import React from 'react'
import styled from '@emotion/styled'

import LeftNav from './LeftNav'
import RightNav from './RightNav'


const RootHeader = styled.header`
  display: flex;
  justify-content: space-between;
  padding: 5px 20px 30px;
  width: 100%;
`

const menuItems = [
  { url: '/', label: 'Home' },
]

class AppNav extends React.PureComponent {
  render() {
    return (
      <RootHeader>
        <LeftNav />
        <RightNav
          menuItems={menuItems}
        />
      </RootHeader>
    )
  }
}

export default AppNav
