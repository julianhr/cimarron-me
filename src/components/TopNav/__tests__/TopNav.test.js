import React from 'react'
import { mount } from 'enzyme'
import { create } from 'react-test-renderer'

import TopNav from '../TopNav'
import MockApp from '~/__tests__/__mocks__/MockApp'


const testProps = {
  menuItems: [
    { url: '/one/path', label: 'label 1', isLinkRouted: false },
  ]
}

const renderApp = (props) => create(<MockApp><TopNav {...props} /></MockApp>)

describe('TopNav', () => {
  describe('props', () => {
    it('requires menuItems', () => {
      const props = {...testProps}
      delete props.menuItems
      expect(() => renderApp(props)).toThrow()
    })
  })

  it('should match snapshot', () => {
    const wrapper = mount(<MockApp><TopNav {...testProps} /></MockApp>)
    expect(wrapper).toMatchSnapshot()
    wrapper.unmount()
  })
})
