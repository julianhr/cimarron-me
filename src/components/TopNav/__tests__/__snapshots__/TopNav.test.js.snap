// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNav should match snapshot 1`] = `
Array [
  <ForwardRef(render)
    styles={
      Object {
        "map": undefined,
        "name": "1n3w7e7-globalStyles-globalStyles",
        "next": undefined,
        "styles": "body{color:#222222,lineHeight 1.3em;font-size:18px;}input,textarea{border:2px solid #BBB;background:#f7f7f7;border-radius:5px;font-size:20px;}input,textarea{border:1px solid #BBB;background:#f7f7f7;font-size:1.1em;padding:4px 8px;}p,h1,h2,h3,h4,h5,h6,figure{margin:0;}h1,h2,h3,h4,h5,h6{color:#9B1518;}p{padding:.8em 0;line-height:1.4em;}h1{font-size:36px;margin:50px 0 40px;}h2{line-height:1.4em;font-size:30px;font-weight:600;margin:40px 0 30px;}h3{font-size:24px;font-weight:600;margin:30px 0 30px;}h4{line-height:1.3em;font-size:18px;}h5{font-size:12px;}h6{font-size:8px;}code{color:#D90000;font-size:14;}label:globalStyles;label:globalStyles;",
      }
    }
  />,
  <RootHeader>
    <header
      className="css-13jirna-RootHeader e1isgil50"
    >
      <Root>
        <div
          className="css-14jo2cv-Root e1r6iq2i0"
        >
          <H3>
            <h3
              className="css-16ij0sm-H3 e1r6iq2i1"
            >
              Julian Hernandez
            </h3>
          </H3>
          <H6>
            <h6
              className="css-mz7w21-H6 e1r6iq2i2"
            >
              Full Stack Software Developer
            </h6>
          </H6>
        </div>
      </Root>
      <Root>
        <div
          className="css-wjqisa-Root e1x2x8qm0"
        >
          <Root>
            <div
              className="css-u96gbo-Root e1uo4w6s0"
            >
              <EmotionCssPropInternal
                __EMOTION_LABEL_PLEASE_DO_NOT_USE__="AnchorTag"
                __EMOTION_TYPE_PLEASE_DO_NOT_USE__="a"
                css={[Function]}
                href="/one/path"
                rel="noopener noreferrer"
                target={null}
              >
                <a
                  className="css-1566xl0-STYLES-STYLES"
                  href="/one/path"
                  rel="noopener noreferrer"
                  target={null}
                >
                  label 1
                </a>
              </EmotionCssPropInternal>
            </div>
          </Root>
        </div>
      </Root>
    </header>
  </RootHeader>,
]
`;
